### PR TITLE
fix general typos and componentDidUpdate in subscription-hoc

### PR DESCRIPTION
**Pull request by frankie303**
### Checklist:

- [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

File: src/react/hoc/subscription-hoc.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ export function withSubscription<
         this.state = { resubscribe: false };
       }
 
-      componentDidUpate(prevProps: TProps) {
+      componentDidUpdate(prevProps: TProps) {
```
Comment (frankie303) [inline src/react/hoc/subscription-hoc.tsx:66]: this seems more than a typo fix. Didn't understand how did this even work before?

Comment (brainkim) [reply src/react/hoc/subscription-hoc.tsx:66]: Jeez that’s odd.

Comment (benjamn) [reply src/react/hoc/subscription-hoc.tsx]: I also had trouble believing nobody had noticed this before, even if `withSubscription` is a legacy API that not many people use, so I went digging, and I found #7506, which the author had closed in frustration when it wasn't reviewed in a timely fashion (my fault). Fortunately, the author also tracked down the solution to the test failure!

I've merged that PR into `main`, and rebased this PR. Once the tests pass (🤞), I'll merge this PR.

Thanks for taking the time to find and fix all these typos @frankie303!

**Review by benjamn — APPROVED** [review]

File: src/react/hoc/subscription-hoc.tsx (hunk 1 above)
Comment (frankie303) [reply src/react/hoc/subscription-hoc.tsx]: Thanks Ben! Glad to hear that it was helpful :)

**Pull request merged**